lib/extole/index.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+/**
+* Module dependencies.
+*/
+
+var bind = require('bind');
+var each = require('each');
+var integration = require('analytics.js-integration');
+var formatConversionParams = require('./lib/formatConversionParams');
+var createConversionTag = require('./lib/createConversionTag');
+
+/**
+* Expose `Extole` integration.
+*/
+
+var Extole = module.exports = integration('Extole')
+  .global('extole')
+  .option('clientId', '')
+  .mapping('events')
+  .tag('main', '<script src="//tags.extole.com/{{ clientId }}/core.js">');
+
+/**
+* Initialize.
+*
+* @param {Object} page
+*/
+
+Extole.prototype.initialize = function(){
+  if (this.loaded()) return this.ready();
+  this.load('main', bind(this, this.ready));
+};
+
+/**
+* Loaded?
+*
+* @return {Boolean}
+*/
+
+Extole.prototype.loaded = function(){
+  return !!(window.extole);
+};
+
+/**
+* Track.
+*
+* @param {Track} track
+*/
+
+Extole.prototype.track = function(track){
+  var user = this.analytics.user();
+  var traits = user.traits();
+  var userId = user.id();
+  var email = traits.email;
+
+  if (!userId && !email) {
+    return this.debug('User must be identified before `#track` calls');
+  }
+
+  var event = track.event();
+  var events = this.events(event);
+
```

Comment:
Using singulars:plurals for instances:collections is generally a good/clear naming scheme, but I see what you mean in that `event` is the Segment event and `events` is a list of mapped Extole events. I'll clarify!